src/partial-templates/partial-templates.service.ts

```diff
@@ -16,33 +16,49 @@ export class PartialTemplatesService {
    * @param {CreatePartialTemplateDto} createPartialTemplateDto - The partialTemplate data to register
    * @returns {Promise<PartialTemplate>} - The registered partialTemplate
    */
-  async create(createPartialTemplateDto: CreatePartialTemplateDto): Promise <PartialTemplate> {
-    // Checks if a partial template with the same name already exists
-    const existingPartialTemplate =  await this.prisma.partialTemplate.findFirst({
-      where: {
-        name: createPartialTemplateDto.name,
-      },
-    });
-    // Throws an exception if a template with the same name exists
-    if (existingPartialTemplate){
-      throw new BadRequestException('A partial template with this name already exists');
-    }
-
-    // Creates a new partial template
-    return this.prisma.partialTemplate.create({
-      data: {
-        ...createPartialTemplateDto,
-      } as any,
-    });
-  }
+    async create(createPartialTemplateDto: CreatePartialTemplateDto): Promise <{ message: string | null; error: { message: string } | null; data: PartialTemplate | null }> {
+      try {
+      await this.validateForeignKeys(createPartialTemplateDto);
+      await this.validateTotalByPosition(createPartialTemplateDto);
+      const newPartialTemplate = await this.prisma.partialTemplate.create({
+        data: {
+          ...createPartialTemplateDto,
+        },
+      });
+  
+      return { message: 'Listo, enviado', error: null, data: newPartialTemplate};
+    } catch (error) {
+      return { message: 'Error al enviar', error: error.message , data: null };
+    }
+ }
 
   /**
-   * Lists all partialTemplates
-   * @returns {Promise<PartialTemplate[]>} - All registered partialTemplates
-   */
-  async findAll(): Promise <PartialTemplate[]> {
-    return this.prisma.partialTemplate.findMany();
-  }
+   * Lists partialTemplates filtered by status
+   * @param {string} [status] - Status to filter partial templates by. Allowed values
+   * @returns {Promise<{name: string, totalHours: number, status: string}[]>} - All registered partialTemplates
+  */
+  async findAll(status?: string): Promise <{name: string, totalHours: number, status: string }[]> {
+    const allowedStatuses = ['pendiente', 'aprobado', 'corrección'];
+    const filter: any = {};
+
+    if(status && allowedStatuses.includes(status)){
+      filter.status = status;
+    }
+
+    const partialTemplates = await this.prisma.partialTemplate.findMany({
+      where: filter,
+      select: {
+        name: true,
+        total: true,
+        status: true,
+      },
+  });
+  return partialTemplates.map(template => ({
+    name: template.name,
+    totalHours: template.total,
+    status: template.status,
+  }));
```

Comment:
Why this is this map function used? just for changing the name of the templates?
keep in mind that there are going to be one partial template for every teacher.
It is worth to perform a map for every partial template just to change the name of the field?